src/api.rs

```diff
@@ -39,6 +38,18 @@ pub struct GoApi {
     pub vtable: GoApi_vtable,
 }
 
+// Debug helper - where to put this?
+pub struct HexBytes<'a>(&'a [u8]);
+
+impl<'a> std::fmt::Display for HexBytes<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        for byte in self.0 {
+            write!(f, "{:02X}", byte)?;
+        }
+        Ok(())
+    }
+}
```

Comment:
We could use the `hex` crate and `hex::encode_upper(canonical)` instead of maintaining the code here

Reply:
Done